### DISCUSSION TITLE
Stacked Bar Chart - Unit Label is accumulative

### DISCUSSION THREAD

**Discussion by Shadowfita** [mantinedev/mantine#5954] [Q&A] [answered]
Hi,

I am trying to use the stacked bar charts and display the data units / counts for each stacked section. The labels are visible but for each section, the number is a sum of the previous section, rather than the distinct amount. 

I have seen this issue discussed on the recharts github page, and the solution seems to be providing a LabelList with a valueAccessor that returns the distinct value, rather than summed value, for each rechart bar component.

Unfortunately with the current implementation in Mantine, I don't see a way to provide any children to the "Bars" that are created in the BarChart component. [BarChart Component Line 181](https://github.com/mantinedev/mantine/blob/e3e3bb834de1f2f75a27dbc757dc0a2fc6a6cba8/packages/%40mantine/charts/src/BarChart/BarChart.tsx#L181)

`<BarChart
              h={250}
              w="100%"
              data={data}
              dataKey="Domain"
              type="stacked"
              tooltipAnimationDuration={200}
              series={[
                { name: '1', color: 'Navy.6' },
                { name: '2', color: 'Yellow.6' },
                { name: '3', color: 'Blue.6' },
              ]}
              barProps={{
                label: {
                  fill: 'white',
                  fontSize: 16,
                },
              }}
              tickLine="y"
              withLegend
            />`

**Comment by rtivital** [comment, marked as answer]
You can pass props down to the `Bar` component with `barProps`

**Comment by Shadowfita** [comment]
Thanks @rtivital. I spotted the function call for barProps in the code.

For others who wish to show distinct values, this is working for me.


```
const valueAccessor =
    (attribute: string) =>
    ({ payload }: { payload: any }) => {
      return payload[attribute];
    };
```

```
              barProps={(item: any) => {
                return {
                  children: <LabelList fill="white" valueAccessor={valueAccessor(item.name)} />,
                };
              }}
```
